include/z64actor.h

```diff
@@ -254,6 +254,10 @@ if neither of the above are set : blue
 0x2000 : translucent, else opaque
 */
 
+#define DYNA_TRANSFORM_NONE 0 // Actors on top are unaffected by movements of the dynapoly actor.
+#define DYNA_TRANSFORM_POS (1 << 0) // Position of the actors on top follows the dynapoly actor's movement.
+#define DYNA_TRANSFORM_ROT_Y (1 << 1) // The Y rotation of the actors on top follows the dynapoly actor's Y rotation.
```

Comment:
Problem I have with these names is that I think DYNA_TRANSFORM doesn't quite relay the bigger picture idea behind the property being set. To me, the intent here is to configure how the actor behaves as a platform

Rough thought for what to name these is something like

``` DYNA_PLATFORM_NO_STICK = 0
 DYNA_PLATFORM_MOVING = 1
 DYNA_PLATFORM_ROTATE = 2
 DYNA_PLATFORM_MOVING_AND_SPINNING = 3
```

Reply:
Yeah `(1 << 1)` by itself is essentially useless unless you want your actor to look jank, which makes it hard to name at all. Looking at those flags from the perspective of how it affects other actors (position and Y rotation) makes it easier to name them since it's pretty clear what happens then, even if it may not make sense.

`DYNA_PLATFORM_MOVING` doesn't really work because when the dynapoly rotates (around any axis), the corresponding function is also responsible for moving the top-actor at the right spot, accounting for all transforms and not just translation of the dyna actor

Also you would naturally describe a stationary rotating platform as "rotating platform", so you'd want to use simply `DYNA_PLATFORM_ROTATE`, but that wouldn't work as expected. This could be solved by not having a define for `(1 << 1)` at all and only for `(1 << 0) | (1 << 1)` but I'm not a fan

So all in all I think it's less sketchy to name the flags by how they affect other actors than trying to describe what the dynapoly actor is

Reply:
Agree that DYNA_PLATFORM_MOVING was an inaccurate suggestion. Still I feel it would be more intuitive to describe things in terms of platform behavior over the more mathy/vague "transformFlags". It's basically implementing the same properties as the Snakes in [Yoshi's Story](https://youtu.be/SeyrQ2zZe-g?t=360). Type 1 and 3 are like the 3D equivalent of the green snake platforms, which move Yoshi as he runs along the top, while type 0 is like the purple snakes that still support the player but don't transport them. I imagine there is some sort of existing terminology for these two styles of platforms, but I can't seem to find anything.

Reply:
the terminology we have for now is `transform_none`, `transform_pos`, `transform_rot` (see you're thinking of how the platform affects whatever's on top now :p)

Reply:
the existing code ONLY affects things that are "on top", noting that ledge grabbing is implemented as being attached to a floor.

Reply:
I'm sorry but it's been a long time, could you elaborate on what you're reacting to?